koku/koku/trino_database.py

```diff
@@ -138,20 +137,23 @@ def executescript(trino_conn, sqlscript, *, params=None, preprocessor=None):
             else:
                 stmt, s_params = p_stmt, params
 
-            try:
-                cur = trino_conn.cursor()
-                cur.execute(stmt, params=s_params)
-                results = cur.fetchall()
-            except TrinoQueryError as trino_exc:
-                trino_statement_error = TrinoStatementExecError(
-                    statement=stmt, statement_number=stmt_num, sql_params=s_params, trino_error=trino_exc
-                )
-                LOG.warning(f"{trino_statement_error!s}")
-                raise trino_statement_error from trino_exc
-            except Exception as exc:
-                LOG.warning(str(exc))
-                raise
-
-            all_results.extend(results)
+            attempts_left = max(attempts_left, 1)
+            for i in range(attempts_left):
+                try:
+                    cur = trino_conn.cursor()
+                    cur.execute(stmt, params=s_params)
+                    results = cur.fetchall()
+                    LOG.info(f"executed trino sql statement {stmt_num}")
+                    all_results.extend(results)
+                    break
+                except TrinoQueryError as trino_exc:
+                    trino_statement_error = TrinoStatementExecError(
+                        statement=stmt, statement_number=stmt_num, sql_params=s_params, trino_error=trino_exc
+                    )
+                    LOG.warning(f"{trino_statement_error!s}")
+                    raise trino_statement_error from trino_exc
+                except Exception as exc:
+                    LOG.warning(str(exc))
+                    raise
```

Comment:
I'm not sure the attempt_left has any retry logic here, so we are just going to raise on the first exception no?

Reply:
Yes it wasn't right. I've put the right logic now.